packages/cpp-debug/src/browser/memory-provider/memory-provider.ts

```diff
@@ -66,26 +66,29 @@ export class MemoryProviderImpl implements MemoryProvider {
 
     /**
      * Read the memory for the given parameters.
-     * @param location the location.
-     * @param length the length.
+     * @param memoryReference the location.
+     * @param count the length.
      * @param offset the offset.
      */
-    async readMemory(location: string, length: number, offset = 0): Promise<Interfaces.MemoryReadResult> {
+    async readMemory(memoryReference: string, count: number, offset = 0): Promise<Interfaces.MemoryReadResult> {
         const session = this.debugSessionManager.currentSession;
         if (session === undefined) {
             throw new Error('No active debug session.');
         }
 
-        const result = await session.sendCustomRequest('cdt-gdb-adapter/Memory', {
-            address: location,
-            length,
+        // @ts-ignore /* Theia 1.17.0 will include the readMemoryRequest in its types. Until then, we can send the request anyway */
+        const result = await session.sendRequest('readMemory', {
+            memoryReference,
+            count,
             offset,
         });
 
-        const bytes = hex2bytes(result.body.data);
-        const address = hexStrToUnsignedLong(result.body.address);
-
-        return { bytes, address };
+        if (result.body?.data) {
+            const bytes = hex2bytes(result.body.data);
+            const address = hexStrToUnsignedLong(result.body.address);
```

Comment:
The `address` property is described by the [DAP docs](https://microsoft.github.io/debug-adapter-protocol/specification#Requests_ReadMemory) as 

> The address of the first byte of data returned.
> Treated as a hex value if prefixed with '0x', or as a decimal value otherwise.

Do you think it might make sense to also manage the case the address is represented as a decimal number to ensure full compatibility?